s3-connector/unittest_s3-connector.py

```diff
@@ -20,62 +20,96 @@ def setUpClass(self):
         self.secret= self.test_cfg['secret']
         self.s3_endpoint=self.test_cfg['s3_endpoint']
         self.region_name=self.test_cfg['region_name']
-        # Intialize connection for S3
+       
         self.s3_connection = s3connector.S3(
             self.test_cfg["key"], 
             self.test_cfg["secret"], 
             self.test_cfg["s3_endpoint"], 
             self.test_cfg["region_name"]
             )
 
-    # Test connection for S3
+    # test connection for s3 resource
     def test_connection(self):
         self.assertTrue(self.s3_connection)
+        
+    #  test connection with incorrect keys
+    def test_conn_with_incorrectkey(self):
+        try:
+          fun = s3connector.S3( 
+            self.test_cfg["incorrect_key"],
+            self.test_cfg["incorrect_secret"],
+            self.test_cfg["s3_endpoint"],
+            self.test_cfg["region_name"]
+            ).list_buckets()
+          for bucket in fun:
+            print(bucket)
+        except ClientError as e:
+            print("Could not connect to s3 resource, check your parameters:",e.response["Error"]["Code"])
+            error = e.response["Error"]["Code"]
+        self.assertEqual(error,"InvalidAccessKeyId")
 
-    # Test list buckets from s3 
+    # test list buckets from s3 resource
     def test_list_bucket(self):
         conn = self.s3_connection.list_buckets()
-        for bucket in conn:
-            print(bucket.name)
-        self.assertTrue(conn)
-        
-    # Test list objects from s3 Bucket
+        for buk in conn:
+            self.assertGreater(len(buk.name), 0)
+           
+  
+    # test non_existing list buckets
+    def test_list_nonexisting_bucket(self):
+        try:
+          conn = self.s3_connection.list_objects(self.test_cfg['bucket_name2'])
+          for b in conn:
+            print(b)
+        except ClientError as e:
```

Comment:
Use the exception test in the above example. 
You could add test for the message in addition to the exception test